wasm-chain-integration/src/lib.rs

```diff
@@ -337,9 +355,105 @@ impl<'a> HasCommon for ReceiveHost<'a> {
 
     fn param(&self) -> &[u8] { &self.param }
 
-    fn metadata(&self) -> &ChainMetadata { &self.receive_ctx.metadata }
+    fn metadata(&self) -> &Self::MetadataType { self.receive_ctx.metadata() }
+
+    fn policies(&self) -> ExecResult<&Self::PolicyType> { self.receive_ctx.sender_policies() }
+}
+
+pub trait HasInitContext {
```

Comment:
I think we need to add some documentation behind why we have this trait and refer to the implementations for why we need it.